p2p/src/net/libp2p/discovery/mdns.rs

```diff
@@ -38,9 +39,15 @@ pub enum Mdns {
 }
 
 impl Mdns {
-    pub async fn new(enabled: bool) -> Self {
-        if enabled {
-            match mdns::Mdns::new(Default::default()).await {
+    pub async fn new(config: &config::MdnsConfig) -> Self {
+        if config.enable_mdns {
```

Comment:
This can still be improved.

How about an enum that looks like this?

```
enum Mdns {
    Enabled(MdnsConfig),
    Disabled,
}
```

Reply:
It doesn't make sense to me, how would you implement this? The reason why it's implemented like it is right now is that it's transparent to the `DiscoveryManager`, meaning the interface for this discovery behavior and for all others is the same: they all implement `poll()` which the `DiscoveryManager` calls irrespective of whether the discovery is actually enabled or not. That information is relayed to the the specified behavior via the config and it internally sets its state as either `Enabled` or `Disabled`.

I'm trying to imagine the implementation for what you're proposing and at least I am having trouble seeing a clean solution.